### DISCUSSION TITLE
When skywalking-go is set to a low sampling rate, such as: sample (0.5), there is a problem of sharing ctx (GetGLS()).  You need to manually call tracing.CleanContext() in AfterInvoke.

### DISCUSSION THREAD

**Discussion by Donghui0** [apache/skywalking#11748] [General]
If multiple HTTP requests share the same goroutine, will the previous ctx be obtained?
for example:
First request: **Unsampled**
GET /hello (entrySpan) (NoopSpan)
             |
     GET /h1 (exitSpan) (NoopSpan)

Second request:
There may be the same goroutine, and the createEntrySpan() -> t.createNoop() -> getTracingContext() method retrieves the **NoopSpan** stored in the first request.

Currently, the tracing.CleanContext() call is added to the afterInvoke method in the gin plugin. 
Is there any problem with doing this? Why do you need to call this method explicitly?

**Comment by wu-sheng** [comment]
@mrproliu Could you check this?

**Reply by mrproliu** [reply]
Currently, For the NoopSpan, there is no reference count yet and GLS continues to be occupied. this is a bug.

**Reply by mrproliu** [reply]
I will try to fix it recently.

**Reply by wu-sheng** [reply]
This should be fixed by https://github.com/apache/skywalking-go/pull/161

**Comment by wu-sheng** [comment]
From my understanding, the context could exit automatically once all spans are closed.
